### DISCUSSION TITLE
Custom Resource Usage That Add EBS Volume/Usage to AWS Instance

### DISCUSSION THREAD

**Discussion by lanandra** [infracost/infracost#2151] [Q&A]
Hi Folks,

Recently me and teammates are exploring Infracost as our infrastructure cost automation.

We have explored some of the features, but found some interesting findings. Below some of the story that we found:
1. We are deploying some Amazon EC2 resources with custom AMI that we've created. That custom AMI consist of 4 EBS including the root block
2. When we run infracost, it is only generating cost of the root block device with 8 GB storage. Although the real infrastructure is using 4 block device with difference storage size. (We assume because in our module, there are no block device arguments in our terraform code. But still we assume because we are using our custom AMI)
3. Tried to explore infracost usage based feature. But in the yaml manifest, we cannot found some block device argument that we can adjust. For example:

```
 # module.our_module_name.module.ec2_instance.aws_instance.this[0]:
    # operating_system: linux # Override the operating system of the instance, can be: linux, windows, suse, rhel.
    # reserved_instance_type: "" # Offering class for Reserved Instances, can be: convertible, standard.
    # reserved_instance_term: "" # Term for Reserved Instances, can be: 1_year, 3_year.
    # reserved_instance_payment_option: "" # Payment option for Reserved Instances, can be: no_upfront, partial_upfront, all_upfront.
    # monthly_cpu_credit_hrs: 0 # Number of hours in the month where the instance is expected to burst. Only applicable with t2, t3 & t4 Instance types. T2 requires credit_specification to be unlimited.
    # vcpu_count: 0 # Number of the vCPUs for the instance type. Only applicable with t2, t3 & t4 Instance types. T2 requires credit_specification to be unlimited.
    # monthly_hrs: 730.0 # Monthly number of hours the instance ran for.
```

Is there any possibility to adjust the usage based of EBS ?

Thanks!

**Comment by tim775** [comment]
Hi @lanandra, unfortunately there is not a way to add ebs costs without specifying them in terraform.  We have [had some discussion](https://github.com/infracost/infracost/issues/1525) in the past about generalizing the usage file so basically any parameter could be added there, but that isn't on our near term roadmap.  It would be good if you could put a 👍 on #1525, we look at those interactions to gauge interest from the community.

**Reply by lanandra** [reply]
Hi @tim775,

Thanks for the information. I wasn't aware that this question has been mentioned in the issue section.
But anyway, i've :+1: that issue and hopefully it can be one of the roadmap in the future :smile: 

Thanks!
